### PR TITLE
Refactor for wide dates csv

### PR DESCRIPTION
**Pull request by TomGoBravo**
### This PR
* Changes `update_data_public_head` to always get data as arguments, making dependency explicit. Also I want `cli.data.update` to have the data so it can write the wide date csv in my next PR.
* Factor out `fips_index_geo_data` so it can be reused
* in `timeseries_test.py` give every row a unique FIPS value
* move `generate_field_summary` so it can be imported by `TimeseriesDataset` without creating an import loop.

### Please Check
- [ ] Are the [JSON schemas](https://github.com/covid-projections/covid-data-model/tree/master/api/schemas) updated?
- [ ] Are the [api docs](https://github.com/covid-projections/covid-data-model/blob/master/api/README.V1.md) updated?
- [ ] Are tests passing?

### CODE REVIEW

File: cli/data.py [hunk 1]
```diff
@@ -34,10 +34,16 @@ def _save_field_summary(timeseries_dataset: TimeseriesDataset, output_path: path
...
 
-    dataset = timeseries_pointer.load_dataset()
-    _save_field_summary(dataset, path_prefix / summary_filename)
+    latest_dataset = combined_datasets.build_us_latest_with_all_fields()
+    timeseries_dataset = combined_datasets.build_us_timeseries_with_all_fields()
+    _, timeseries_pointer = combined_dataset_utils.update_data_public_head(
+        path_prefix, latest_dataset, timeseries_dataset
+    )
+
+    if summary_filename:
```
Comment (ghop02) [inline cli/data.py:44]: There will always be a summary filename, correct?

File: libs/datasets/combined_datasets.py [hunk 2]
```diff
@@ -293,23 +289,7 @@ def _build_data_and_provenance(
...
-    all_identifiers = pd.concat(
-        df.reset_index().loc[
-            :, [CommonFields.FIPS] + list(df.columns.intersection(preserve_columns))
-        ]
-        for df in datasource_dataframes.values()
-    ).drop_duplicates()
-    # Make a DataFrame with a unique FIPS index. If multiple rows are found with the same FIPS then there
-    # are rows in the input data sources that have different values for county name, state etc.
-    fips_indexed = all_identifiers.set_index(CommonFields.FIPS, verify_integrity=True)
+    fips_indexed = fips_index_geo_data(pd.concat(datasource_dataframes.values()))
```
Comment (ghop02) [inline libs/datasets/combined_datasets.py:292]: Can you call this `dataset_utils.fips_index_geo_data`?

File: cli/data.py (hunk 1 above)
Comment (TomGoBravo) [reply cli/data.py:44]: Not with `./run.py data update --summary-filename=''`. I wanted a way to disable it to see if the wide-dates output is faster.

File: libs/datasets/combined_datasets.py (hunk 2 above)
Comment (TomGoBravo) [reply libs/datasets/combined_datasets.py]: Done in 5c7c0f0

**Pull request merged**